### PR TITLE
feat: add a runtime for FrankenPHP with Symfony

### PR DESCRIPTION
**Pull request by dunglas**
Add a runtime for the worker mode of [FrankenPHP](https://frankenphp.dev). To be released this afternoon at ForumPHP!

frankenphp

### CODE REVIEW

File: src/frankenphp-symfony/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# RoadRunner Runtime for Symfony
```
Comment (alexander-schranz) [inline src/frankenphp-symfony/README.md:1]:
```suggestion
# FrankenPHP Runtime for Symfony
```

**Review by alexander-schranz — CHANGES_REQUESTED** [review]
small copy & paste error. Looking forward to Early Hints 👀

File: src/frankenphp-symfony/src/Runner.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                $sfRequest = Request::createFromGlobals();
+                $sfResponse = $this->kernel->handle($sfRequest);
+
+                $sfResponse->send();
+            });
+
+            if ($this->kernel instanceof TerminableInterface && $sfRequest && $sfResponse) {
+                $this->kernel->terminate($sfRequest, $sfResponse);
+            }
+        } while ($ret);
```
Comment (Guikingone) [inline src/frankenphp-symfony/src/Runner.php:40]: Sorry for the review but maybe the callable could define its return type `: void` if I'm right, same for the first one: 

```php
$server = array_filter($_SERVER, static fn (string $key): bool => !str_starts_with($key, 'HTTP_'), ARRAY_FILTER_USE_KEY);
```

I'm not sure if the current lib tend to define return types on callable 🙁

File: src/frankenphp-symfony/composer.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    "license": "MIT",
+    "type": "library",
+    "authors": [
+        {
+            "name": "Kévin Dunglas",
+            "email": "kevin@dunglas.dev"
+        }
+    ],
+    "require": {
+        "php": ">=8.2.0",
```
[the remark this answers is not in the document]
Comment (dunglas) [reply src/frankenphp-symfony/composer.json:13]: Even FrankenPHP only ships 8.2, this code also works on 8.1 (and even with another SAPI). I can relax the version if you want.

[the remark this answers is not in the document]
Comment (Nyholm) [reply src/frankenphp-symfony/composer.json:13]: Im fine with this

File: src/frankenphp-symfony/src/Runner.php (hunk 2 above)
Comment (Nyholm) [reply src/frankenphp-symfony/src/Runner.php:40]: I usually dont add return types on callables. It gives zero to little benefit. But it is just a matter of taste. As long as CI is not complaining, Im fine with adding return types.

**Review by Nyholm — APPROVED** [review]

File: src/frankenphp-symfony/src/Runner.php [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+                $sfRequest = Request::createFromGlobals();
+                $sfResponse = $this->kernel->handle($sfRequest);
+
+                $sfResponse->send();
+            });
+
+            if ($this->kernel instanceof TerminableInterface && $sfRequest && $sfResponse) {
+                $this->kernel->terminate($sfRequest, $sfResponse);
+            }
```
Comment (alexander-schranz) [inline src/frankenphp-symfony/src/Runner.php:39]: Should this part not be also inside the `frankenphp_handle_request`. Or does it have no difference here? If its inside or outside the callable. I'm not sure if the terminate is this way called for every request as it should, else I know some projects which would crash into unexpected state.

File: src/frankenphp-symfony/composer.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (alexander-schranz) [reply src/frankenphp-symfony/composer.json:13]: `8.0` as we currently still have that in the CI would be nice :)

```suggestion
        "php": ">=8.0.0",
```

File: src/frankenphp-symfony/src/Runner.php (hunk 4 above)
Comment (Nyholm) [reply src/frankenphp-symfony/src/Runner.php:39]: Frankenphp only cares about putting out the response. A "cleanup" or background process could be running outside IMHO.

**Review by alexander-schranz — APPROVED** [review]

**Comment by Nyholm** [comment]
Could you please rebase your PR now when #125 is merged?

File: src/frankenphp-symfony/src/Runner.php (hunk 4 above)
Comment (dunglas) [reply src/frankenphp-symfony/src/Runner.php:39]: > I'm not sure if the terminate is this way called for every request as it should

It does.

**Comment by dunglas** [comment]
@Nyholm rebased

**Comment by Nyholm** [comment]
Thank you!

**Comment by Nyholm** [comment]
https://packagist.org/packages/runtime/frankenphp-symfony

**Comment by dunglas** [comment]
Thank you very much @Nyholm!

**Pull request merged**